### ISSUE TITLE
Naming issues

### ISSUE THREAD

**Issue by bastimeyer** [streamlink/streamlink#65]
We should discuss the project's naming conflicts.

Since the creation of the fork, many people ([including myself](https://github.com/bastimeyer/livestreamer-twitch-gui/issues/289#issuecomment-247846424)) have mentioned potential issues with the name "streamlink". The reason for this is an already existing organization called [StreamLink Software](http://www.streamlinksoftware.com/about), whose name stands in conflict with this project here.

I'd like to see this issue being solved before it is getting too late. The size and adoption of the fork is still relatively low, so finding a new conflict-free name shouldn't cause too much damage. The later we wait for this, the worse it gets.

**Comment by cdrage** [comment]
Imo, I think it's okay and there's no issue. Two different products. Sort of like how livestreamer sounds the same as livestream.

There's no sign of usage from them on github.

If anyone can come up with a better name feel free to speak up!

**Comment by MrLint** [comment]
I'm startign to kick around names in my head until i find something i like.

**Comment by gravyboat** [comment]
I'm not too concerned. We could always reach out to them to see if it's a problem since we're not exactly getting rich off of working on the project. Any name related to streaming like this is most likely going to be taken already so I would rather focus our energies into the software rather than worrying about potential legal issues. @cdrage Should we shoot them a quick email about this? If so we should create a separate issue and ensure they know the email will be publicly posted for project transparency purposes.

**Comment by alphapapa** [comment]
A few ideas:

* streamhelper
* streamfetcher
* streamloader

And a silly one: streamerer

**Comment by MrLint** [comment]
Streamverter?

**Comment by cdrage** [comment]
@gravyboat I think we should stay with just streamlink. Honestly, neither projects relate to each-other (one is for streaming, other is for management tasks / task delegation).

They've got our contact information, but it won't affect us in regards to trademark law.

Like what @gravyboat mentioned, regardless of what we pick it's most likely going to be taken. I bring up the fact that "livestreamer" is still close to the name "livestream" yet they had no problems (and I think it's a 4 year old project?)

**Comment by alphapapa** [comment]
> They've got our contact information, but it won't affect us in regards to trademark law.

You might be right, but you might not be.  Remember that Firefox was once called Phoenix, and then Firebird, and then finally Firefox.  Firebird is the name of some database software--clearly nothing to do with a web browser, yet Mozilla changed the name anyway.  Not that this project would attract attention like Firefox, but food for thought.  :)

**Comment by TheSneakySniper** [comment]
how about firestreamer

**Comment by martinth** [comment]
What about something really short and handy? Because it would actually be the name of the executable so you type it a lot. 

- stream
- strim 
- strm (going full startup here) 
-... 

Something in that direction. 

Additional benefit of having a single English word would be that there is imho no way you could ever run into leagle issues because there is afaik no way to trademark singular words.

**Comment by cdrage** [comment]
@martinth problem with that (other than strm) is that it would be really hard to search for on google / seo for people whom want a streaming client :(

**Comment by MrLint** [comment]
@cdrage So speaking of searching, I had a bit of trouble finding this project because of the all the existing uses of streamlink. I had to go back to the post by @bastimeyer to link into it directly.

**Comment by bastimeyer** [comment]
@gravyboat @cdrage 
*Potential* legal issues should not be ignored. Livestreamer didn't have those issues, because "livestream" is just a normal, generic word and "livestreamer" is not related to any company name.

> Should we shoot them a quick email about this? If so we should create a separate issue and ensure they know the email will be publicly posted for project transparency purposes.

This is what should be done before thinking about renaming the project.

@MrLint 
This is true, but that's just because streamlink is new and doesn't have any refering links of other websites yet. If you search for livestreamer, you will also get results of "Livestream LLC". I think the results can be changed quickly, but there will always remain a certain confusion for a couple of users. Just think about the query autocomplete function of google, suggesting "streamlink software".

Based on @martinth's post:
`strmlnk` is basically the same name, just without vowels. So if we need to rename the project, this would be my suggestion.

**Comment by cdrage** [comment]
@bastimeyer 

They're registered as "StreamLink Software Inc".

I've sent them an email for initial point of contact, once legal has contacted me I'll relay the email publicly here.

**Comment by TheSneakySniper** [comment]
I like linkstream

**Comment by tp0** [comment]
I agree that shorter name would be beneficial, since that is what will be written everytime for the command (that's why I make aliases/bash functions, but that's just 2-3 letters).

`strmlnk` particularly is perhaps hard to type: try typing it 10 times really fast and see how many times you get it wrong.

ps. I always thought think the original name was confusing, because of the "-streamer" part, because it sounds more like screencasting.

**Comment by TheGoblinHero** [comment]
livestreamer-ng or something similar

**Comment by WiniD** [comment]
Why not just call it `livestreamer2`?

**Comment by karlo2105** [comment]
And why don't u use livestreamer name like originál?

**Comment by cdrage** [comment]
@karlo2105 developer abandoned the project

**Comment by karlo2105** [comment]
@cdrage 
But you can fork Livestreamer by keeping the same name of project.
I think it's pointless to change name as it will make several projects of same program with different names.

**Comment by sbstp** [comment]
The name is taken on pypi by the old, unmaintained project. We can't push updates to pypi using the old name.

**Comment by karlo2105** [comment]
You can push changes on existing project by pulling requests. 
If you want to make separate project, I will choose Livestreamer2 name.

**Comment by sbstp** [comment]
You don't understand. We can't update the `livestreamer` package on python's official distribution platform (pypi) because we don't have the permissions, and the maintainer is not responsive.

**Comment by karlo2105** [comment]
Is there any alternative to pypi to keep on with same name project ?

**Comment by gravyboat** [comment]
@karlo2105 No there isn't.

**Comment by karlo2105** [comment]
Some members made fork from livestreamer by keeping same name and contributed to original project. 
How did it work on their side pypi needs permissions ?

**Comment by skulblakka** [comment]
>and contributed to original project. 

That's why. They made Pull Request to the original project that were pulled into it. However as the original maintainer is no longer continuing the project no Pull Request are being merged.

**Comment by lawin1** [comment]
Just add "-er": streamlinker.

**Comment by MrLint** [comment]
Streamera? I dont see this on any search results.

**Comment by TheSneakySniper** [comment]
@lawli3t perfect

**Comment by weedy** [comment]
> strm (going full startup here)

strmr
As in strimmer is your group full? We should go full meme naming

**Comment by strudelkuchen** [comment]
`Livestreamer2` sounds good.
The first major fork of `mplayer` was called `mplayer2`, which then got forked and gave us `mpv`.

**Comment by SanyaIV** [comment]
I second Livestreamer2 but also like Streamera.

**Comment by alphapapa** [comment]
`lvstrmr` is short and seems to not turn up search results.

**Comment by scottbernstein** [comment]
I like streamlink just fine  :)

________________________________

**Comment by bastimeyer** [comment]
@cdrage Did you receive a response from them yet? It's been almost four weeks since you've sent the email. I think this can be interpreted as "we don't care".

**Comment by cdrage** [comment]
@bastimeyer yeah, despite the correspondent forwarding the email to the legal department, i hadn't heard anything back.

```
My apologizes, I do not distribute leads.. simply follow up on them. I’ll get your message to the correct person.
 
Thanks,
 
Brittani Ellis
Sales Representative
StreamLink Software
bellis@streamlinksoftware.com
(o) 216. 800. 4984
 
From: Charlie Drage [mailto:charlie@charliedrage.com] 
Sent: Tuesday, October 4, 2016 10:19 AM
To: Brittani Ellis <bellis@streamlinksoftware.com>
Subject: Re: Follow up to "Contact Us" Request- AmpliFund
```

**Comment by gravyboat** [comment]
@cdrage For visibility purposes could you please post your original email to them as well? Then we have a public record of it.

**Comment by cdrage** [comment]
@gravyboat 
Checked through my email and there automated system / ticket thingy that replies back didn't include my original reply. Basically, I said that we have a project on github called streamlink, that takes online streams and pipes it into a video player.

This is the full email chain:

```
Hi Charlie,
 
I wanted to follow up to your interest in our grant management tool, AmpliFund. Let me know if you have any questions or would like to schedule a demonstration. Feel free to send me some dates/times that work for your schedule and I’d be happy to get an invite sent your way.
 
Best Regards,
 
Brittani Ellis
Sales Representative
(216) 800-4984
bellis@streamlinksoftware.com
812 Huron Road E, Ste 350
Cleveland, OH 44115
cid:image003.png@01D19015.7A83B560

Errrr, I think you sent me an automated message.

Could you please read the initial contact request and forward my information to the legal department?

Many thanks!

...
My apologizes, I do not distribute leads.. simply follow up on them. I’ll get your message to the correct person.
 
Thanks,
 
Brittani Ellis
Sales Representative
StreamLink Software
bellis@streamlinksoftware.com
(o) 216. 800. 4984
```

**Comment by alphapapa** [comment]
Regardless of whether they respond with permission or don't respond at all, I think it would be unwise to start this fork off with a potential "time bomb" that could force a name change at an unknown future date.  They could always change their minds, e.g. if someone else takes over that role at the company.

This is a software project, and that "StreamLink" company makes software, so this is an obvious trademark issue.  Just think of how much of a hassle it would be to get 6 months into the fork and be forced to change its name everywhere: GitHub, PyPI, web sites, notices to the community, transitional packages, etc.

Let's pick a name that doesn't start us off with problems.  :)

**Comment by karlo2105** [comment]
Why not Livestreamer2 ? It would be much easier to merge everything with original project when Chrippa decides to give access to his project. ;)

**Comment by beardypig** [comment]
I think this has been resolved by not doing anything, seems like we are keeping the name `streamlink`...

**Comment by cdrage** [comment]
Let's close this for now :)
